tests/NetEscapades.EnumGenerators.Tests/Snapshots/EnumGeneratorTests.CanGenerateEnumExtensionsWithCustomNamespaceAndName.verified.txt

```diff
@@ -75,23 +76,7 @@ namespace A.B
 #endif
             string? name, 
             out MyTestNameSpace.MyEnum value)
-        {
-            switch (name)
-            {
-                case nameof(MyTestNameSpace.MyEnum.First):
-                    value = MyTestNameSpace.MyEnum.First;
-                    return true;
-                case nameof(MyTestNameSpace.MyEnum.Second):
-                    value = MyTestNameSpace.MyEnum.Second;
-                    return true;
-                case { } s when int.TryParse(name, out var val):
-                    value = (MyTestNameSpace.MyEnum)val;
-                    return true;
-                default:
-                    value = default;
-                    return false;
-            }
```

Comment:
We're no longer special-casing this, I wonder what the impact is on going through the switch-expression with string comparison over the simple switch 🤔

Reply:
What do you mean by special casting? Is it the int parse?

```
case { } s when int.TryParse(name, out var val):
                    value = (MyTestNameSpace.MyEnum)val;
                    return true;

```
String comparison must be slower, as the simple switch is hashed, so it's almost O(1). That's why the `current.Equals(other, Compare.Ordinal)` is slower in the Spanify branch. 
Or you though to something else?

Reply:
> What do you mean by special casing

Sorry, I meant _not_ having the switch for the case-sensitive case seems like a regression (exactly what you described). 

I don't really see a reason to not have the separate `TryParse` and `TryParseIgnoreCase` branches, given we can optimise the `TryParse` case?

Reply:
Optimized the TryParse when the case if not ignored.